walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,126 @@
+      TimeRange nextEventTimeRange = eventList.get(i + 1).getWhen();
+
+      if (!requestHasEventAttendees(request, eventList.get(i))) {
+        eventsSkipped ++;
+        continue;
+      }
+
+      if (currEventTimeRange.contains(nextEventTimeRange)) {
+        // If there is a 3rd event, connect events 1 and 3. Otherwise, connect with the end of day.
+        if (i + 2 < eventList.size() - 1) {
+          TimeRange nextnextEventTimeRange = eventList.get(i + 2).getWhen();
+          availableTimes.add(TimeRange.fromStartEnd(currEventTimeRange.end(),
+            nextnextEventTimeRange.start(), true));
+          i++;
+        } else {
+          availableTimes.add(TimeRange.fromStartEnd(currEventTimeRange.end(),
+            TimeRange.END_OF_DAY, true));
+        }
+      } else {
+          // If current event completely overlaps the next event, there's no reason to check
+          // the next one too, as it's start and end are completely contained.
+          if (currEventTimeRange.overlaps(nextEventTimeRange)) {
+            continue;
+          }
+
+          if (nextEventTimeRange.start() - currEventTimeRange.end() >= request.getDuration()) {
+            availableTimes.add(TimeRange.fromStartEnd(currEventTimeRange.end(),
+              nextEventTimeRange.start(), false));  
+          } 
+        }
+    }
+
+    if (requestHasEventAttendees(request, lastEvent)) {
+      if (availableTimes.size() != 0){
+        if (nothingEndsTheDay(availableTimes, eventList)) {
+          availableTimes.add(TimeRange.fromStartEnd(eventList.get(eventList.size() - 1).getWhen().end(),
+            TimeRange.END_OF_DAY, true));
+        } 
+      } else {
+          if (fitsOnlyAtTheEnd(request, availableTimes, eventList)) {
+              availableTimes.add(TimeRange.fromStartEnd(eventList.get(eventList.size() - 1).getWhen().end(),
+                TimeRange.END_OF_DAY, true));
+            }
+        } 
+    } 
+
+    if (eventsSkipped == eventList.size()) {
+      availableTimes.add(TimeRange.WHOLE_DAY);
+    }
+
+    return availableTimes;
+  }
+
+  static private boolean requestHasEventAttendees(MeetingRequest request, Event currEvent) {
+    List<String> requestAttendees = new ArrayList<>(request.getAttendees());
+    List<String> eventAttendees = new ArrayList<>(currEvent.getAttendees());
+
+    return requestAttendees.stream().anyMatch(eventAttendees::contains);
+  }
+
+  // Check if the first event in our list already includes the start of the day.
+  static private boolean StartOfDayIsFree(List<Event> eventList) {
+    return eventList.get(0).getWhen().start() != TimeRange.START_OF_DAY;
+  }
+
+  // Check if there are no available times already added in our list which contain the end of day
+  // and any events that end the day.
+  static private boolean nothingEndsTheDay(List<TimeRange> availableTimes, List<Event> eventList) {
+    // TimeRange.END_OF_DAY returns 1439 (23*60 + 59), but the actual end of day measured in tests
+    // is 1440.
```

Comment:
so in this case I think it's more appropriate to use `TimeRange.WHOLE_DAY.duration()`